error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/PreconditionsRules.java

```diff
@@ -72,35 +74,73 @@ void after(int index, int size, String message) {
     }
   }
 
-  /** Prefer {@link Preconditions#checkNotNull(Object)} over more verbose alternatives. */
-  static final class CheckNotNull<T> {
+  /** Prefer {@link Objects#requireNonNull(Object)} over non-JDK alternatives. */
+  static final class RequireNonNull<T> {
+    @BeforeTemplate
+    T before(T object) {
+      return checkNotNull(object);
+    }
+
+    @AfterTemplate
+    @UseImportPolicy(STATIC_IMPORT_ALWAYS)
+    T after(T object) {
+      return requireNonNull(object);
+    }
+  }
+
+  /** Prefer {@link Objects#requireNonNull(Object)} over non-JDK alternatives. */
```

Comment:
```suggestion
  /** Prefer {@link Objects#requireNonNull(Object)} over more verbose or non-JDK alternatives. */
```
But, assuming we drop the `checkArgument` check:
```suggestion
  /** Prefer {@link Objects#requireNonNull(Object)} over more verbose alternatives. */
```